hazelcast/src/hazelcast/client/protocol.cpp

```diff
@@ -513,26 +530,21 @@ namespace hazelcast {
                     if (isCompleted) {
                         //MESSAGE IS COMPLETE HERE
                         message->wrap_for_read();
-                        connection.handleClientMessage(std::move(message));
                         isCompleted = true;
 
-                        //TODO: add fragmentation later
-/*
-                        if (message->isFlagSet(ClientMessage::BEGIN_AND_END_FLAGS)) {
+                        if (message->is_flag_set(ClientMessage::UNFRAGMENTED_MESSAGE)) {
                             //MESSAGE IS COMPLETE HERE
                             connection.handleClientMessage(std::move(message));
-                            isCompleted = true;
                         } else {
-                            if (message->isFlagSet(ClientMessage::BEGIN_FLAG)) {
+                            message->drop_fragmentation_frame();
+                            if (message->is_flag_set(ClientMessage::BEGIN_FRAGMENT_FLAG)) {
                                 // put the message into the partial messages list
                                 addToPartialMessages(message);
-                            } else if (message->isFlagSet(ClientMessage::END_FLAG)) {
+                            } else if (message->is_flag_set(ClientMessage::END_FRAGMENT_FLAG)) {
                                 // This is the intermediate frame. Append at the previous message buffer
                                 appendExistingPartialMessage(message);
-                                isCompleted = true;
                             }
```

Comment:
What happens if a message has 3 or more fragments? We are handling messages either with BEGIN_FRAGMENT_FLAG or END_FRAGMENT_FRAGMENT set but what about the middle ones?